src/components/AddressSearch.js

```diff
@@ -100,12 +92,16 @@ const AddressSearch = (props) => {
                 label: props.label,
                 containerStyles: props.containerStyles,
                 errorText: props.errorText,
+                value: props.value,
                 onChangeText: (text) => {
-                    const isTextValid = !_.isEmpty(text) && _.isEqual(text, props.value);
-
-                    // Ensure whether an address is selected already or has address value initialized.
-                    if (!_.isEmpty(googlePlacesRef.current.getAddressText()) && !isTextValid) {
-                        saveLocationDetails({});
+                    // This line of code https://github.com/FaridSafi/react-native-google-places-autocomplete/blob/47d7223dd48f85da97e80a0729a985bbbcee353f/GooglePlacesAutocomplete.js#L148
+                    // will call onChangeText passing '' after the component renders the first time, clearing its value. Why? who knows, but we have to skip it.
```

Comment:
Why wasn't this an issue before?

Reply:
I'm guessing that the:
```

    const googlePlacesRef = useRef();
    useEffect(() => {
        if (!googlePlacesRef.current) {
            return;
        }

        googlePlacesRef.current.setAddressText(props.value);
    }, []);
```

may have been fixing it. I will check that.